### PR TITLE
Python 3 and PyPy support.

### PR DESCRIPTION
**Pull request by jamadden**
100% coverage.

Needs a fix for zopefoundation/zope.catalog#5. We have a local workaround, but only for these tests.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,55 +18,86 @@
...
+          'Programming Language :: Python:: 3.4',
+          'Programming Language :: Python:: 3.5',
+          'Programming Language :: Python:: 3.6',
+          'Programming Language :: Python :: Implementation :: CPython',
+          'Programming Language :: Python :: Implementation :: PyPy',
           'Natural Language :: English',
           'Operating System :: OS Independent',
           'Topic :: Internet :: WWW/HTTP',
-          'Framework :: Zope3'],
+          'Framework :: Zope3'
```
Comment (mgedmin) [inline setup.py]: Add a trailing comma?

(I wouldn't nag, but you seemed receptive in the past.)

File: src/zope/app/catalog/browser/tests.py [hunk 2]
```diff
@@ -13,16 +13,57 @@
...
 
+# XXX: Workaround for https://github.com/zopefoundation/zope.catalog/issues/5
+import zope.schema
+orig_bytesline = zope.schema.BytesLine
+try:
+    zope.schema.BytesLine = zope.schema.NativeStringLine
+    import zope.catalog.text
+    assert hasattr(zope.catalog.text, 'ITextIndex')
+finally:
+    zope.schema.BytesLine = orig_bytesline
```
Comment (mgedmin) [inline src/zope/app/catalog/browser/tests.py]: Reminder to remove once that's been fixed.

**Review by mgedmin — APPROVED** [review]
LGTM

File: setup.py (hunk 1 above)
Comment (jamadden) [reply setup.py]: Whoops, missed that. Thanks!

**Comment by jamadden** [comment]
Thanks for the review! A PyPI release when convenient will be helpful in moving zc.catalog forward.

**Comment by mgedmin** [comment]
You have PyPI rights now.

**Pull request merged**